sys/net/application_layer/dhcpv6/client.c

```diff
@@ -833,6 +833,13 @@ static bool _parse_ia_pd_option(dhcpv6_opt_ia_pd_t *ia_pd)
         if (lease->parent.ia_id.id != ia_id) {
             continue;
         }
+        // Is taking the routable one(s) really *better* than taking the ULA?
+        // The ULA might be what is more stable, or what persists through
+        // renumberings ... but probably picking routable *is* best, and when
+        // that prefix becomes unavailable we'll just wait for a network
+        // reconfiguration to bring everyone up on ULAs instead. (If of course
+        // we had the capacity for multiple prefixes on all devices...).
+        bool take_only_routable = true;
```

Comment:
Can we eventually add a configuration for this?

Reply:
We could, but I'd rather have good defaults primarily. A usable BR should be provided in RIOT to be set up as easily as setting up a WiFi access point -- and that usually doesn't require knowledge of whether or not particular types of addresses are around.

A better version could express some policy: Use as many prefixes as possible, but use globally routable ones first. Or use a globally routable ones first, but if there is a ULA use that too if there is space (because that's expected to be more stable). Maybe better choices in the numbers chosen (adjusting CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX to match the number of prefixes we can use) can be taken, and maybe then (if we only request one prefix) this will push the problem of selecting the right prefix(es) upstream to where more data is available and a better decision can be made.

(Hard thing is: I don't know how most of these numbers play together, and if we can assume any of them to be synchronized)